src/main/groovy/com/jetbrains/python/envs/PythonEnvsExtension.groovy

```diff
@@ -28,58 +27,86 @@ class PythonEnvsExtension {
      * @param linkWithVersion if true, binary will be linked with version name.
      * I.e. "python" will be have "python2.7" link (same for exe file). Used for envs line tox.
      */
-    void conda(final String envName, final String version, final List<String> packages, final boolean linkWithVersion) {
-        def pipPackages = packages.findAll { !it.startsWith(CONDA_PREFIX) }
-        def condaPackages = packages.findAll { it.startsWith(CONDA_PREFIX) }.collect {
-            it.substring(CONDA_PREFIX.length())
-        }
-        condaEnvs << new VersionedPythonEnv(envName, version, pipPackages, condaPackages, linkWithVersion)
+    void conda(final String envName,
+               final String version,
+               final List<String> packages = null,
+               final boolean linkWithVersion = false) {
+        List<String> pipPackages = packages.findAll { !it.startsWith(CONDA_PREFIX) }
+        List<String> condaPackages = packages.findAll { it.startsWith(CONDA_PREFIX) }
+                                             .collect { it.substring(CONDA_PREFIX.length()) }
+        condaEnvs << new CondaEnv(envName, version, pipPackages, condaPackages, linkWithVersion)
     }
 
-    void virtualenv(final String envName, final List<String> packages) {
-        virtualEnvs << new PythonEnv(envName, packages)
+    void python(final String envName, final String version, final List<String> packages = null) {
+        pythonEnvs << new PythonEnv(envName, "python", version, packages)
     }
 
-    String install(final String packageName) {
-        return CONDA_PREFIX + packageName
+    void jython(final String envName, final List<String> packages = null) {
+        pythonEnvs << new PythonEnv(envName, "jython", "", packages)
```

Comment:
For type it's better to pass a enum or create a different class, that will denote a different type (e.g. JythonEnv) and then use instanceof to return the corresponding enum value.